projects/habitat_ovmm/eval_vectorized.py

```diff
@@ -154,67 +160,62 @@ def stop():
+                                **episode_metrics[episode_key],
                             }
+                            episode_metrics[episode_key]["goal_name"] = info[
+                                "goal_name"
+                            ]
                             print(
                                 f"Finished episode {episode_key} after "
                                 f"{round(time.time() - start_time, 2)} seconds"
                             )
 
                     elif num_episodes_per_env is not None:
                         if episode_idxs[e] < num_episodes_per_env:
+                            metrics_at_episode_end = {
+                                f"END." + k: v for k, v in metrics.items()
+                            }
                             episode_metrics[episode_key] = {
-                                **metrics,
-                                "goal_name": info["goal_name"],
+                                **metrics_at_episode_end,
+                                **episode_metrics[episode_key],
                             }
+                            episode_metrics[episode_key]["goal_name"] = info[
+                                "goal_name"
+                            ]
                         episode_idxs[e] += 1
                         print(
                             f"Episode indexes {episode_idxs} / {num_episodes_per_env} "
                             f"after {round(time.time() - start_time, 2)} seconds"
                         )
 
-                        # [temporary] to print running metrics
-                        aggregated_metrics = defaultdict(list)
-                        metrics = set(
-                            [
-                                k
-                                for k in list(episode_metrics.values())[0].keys()
-                                if k != "goal_name"
-                            ]
-                        )
-                        for v in episode_metrics.values():
-                            k = "nav_to_obj_success"
-                            aggregated_metrics[f"{k[:3]}/total"].append(v[k])
-                            aggregated_metrics[f"{k[:3]}/{v['goal_name'][:16]}"].append(
-                                v[k]
-                            )
-
-                        aggregated_metrics = dict(
-                            sorted(
-                                {
-                                    k2: v2
-                                    for k1, v1 in aggregated_metrics.items()
-                                    for k2, v2 in {
-                                        f"{k1}/mean": np.round(np.mean(v1), 2),
-                                    }.items()
-                                }.items()
-                            )
-                        )
-
-                    if len(episode_metrics) % 5 == 0:
+                    if len(episode_metrics) % 1 == 0:
```

Comment:
% 1 should maybe be a parameter somewhere? right now this is writing every time right